src/main/java/org/jabref/logic/bibtexkeypattern/BibtexKeyPatternUtil.java

```diff
@@ -742,18 +748,122 @@ private static String getTitleWordsWithSpaces(int number, String title) {
+     */
+    public static String getCamelizedTitle(String title) {
+        return keepLettersAndDigitsOnly(camelizeTitle(title));
+    }
+
+    private static String camelizeTitle(String title) {
+        StringBuilder stringBuilder = new StringBuilder();
+        String formattedTitle = formatTitle(title);
+
+        try (Scanner titleScanner = new Scanner(formattedTitle)) {
+            while (titleScanner.hasNext()) {
+                String word = titleScanner.next();
+
+                // Camelize the word
+                word = word.substring(0, 1).toUpperCase() + word.substring(1);
+
+                if (stringBuilder.length() > 0) {
+                    stringBuilder.append(' ');
+                }
+                stringBuilder.append(word);
+            }
+        }
+
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Capitalises the significant words of the "title" field in the given BibTeX entry
+     */
+    public static String camelizeSignificantWordsInTitle(String title) {
+        StringBuilder stringBuilder = new StringBuilder();
+        String formattedTitle = formatTitle(title);
+        Boolean camelize;
+
+        try (Scanner titleScanner = new Scanner(formattedTitle)) {
+            while (titleScanner.hasNext()) {
+                String word = titleScanner.next();
+                camelize = true;
+
+                // Camelize the word if it is significant
+                for (String smallWord : Word.SMALLER_WORDS) {
+                    if (word.equalsIgnoreCase(smallWord)) {
+                        camelize = false;
+                        continue;
+                    }
+                }
+                // We want to capitalize significant words and the first word of the title
+                if (camelize || (stringBuilder.length() == 0)) {
+                    word = word.substring(0, 1).toUpperCase() + word.substring(1);
+                } else {
+                    word = word.substring(0, 1).toLowerCase() + word.substring(1);
+                }
+
+                if (stringBuilder.length() > 0) {
+                    stringBuilder.append(' ');
+                }
+                stringBuilder.append(word);
+            }
+        }
+
+        return stringBuilder.toString();
+    }
+
+
+    public static String removeSmallWords(String title) {
+        StringBuilder stringBuilder = new StringBuilder();
+        String formattedTitle = formatTitle(title);
+
+        try (Scanner titleScanner = new Scanner(formattedTitle)) {
+            mainl: while (titleScanner.hasNext()) {
+                String word = titleScanner.next();
+
+                for (String smallWord : Word.SMALLER_WORDS) {
+                    if (word.equalsIgnoreCase(smallWord)) {
+                        continue mainl;
+                    }
+                }
+
+                if (stringBuilder.length() > 0) {
+                    stringBuilder.append(' ');
```

Comment:
Thank you! :+1: We're just trying to improve the code quality wherever possible 😇